### DISCUSSION TITLE
CVE-2023-45288 on 3.5.14

### DISCUSSION THREAD

**Discussion by charlie-ang-collibra** [etcd-io/etcd#18168] [Q&A] [answered]
Hi!   I'm unfamiliar with posting to this forum, so please bear with me if there are rules that I fail to follow 🙏 

I am looking to address CVE-2023-45288 for my organization that is discovered by Snyk when scanning etcd 3.5.13.  

My understanding (from https://github.com/etcd-io/etcd/issues/17703) was that `golang/x/net` was upgraded to 0.23.0 as per the merge to go.mod and go.sum in https://github.com/etcd-io/etcd/pull/17708/files).  These changes were supposedly released in 3.5.14

However, when I try to use etcd 3.5.14 using the bitnami/etcd or the coreos/etcd docker images, snyk continues to find version 0.17.0 

![image](https://github.com/etcd-io/etcd/assets/96485235/32b0c78b-afb5-49d7-b3f7-02927cf353f4)

When I download the actual binary release from the `https://storage.googleapis.com/etcd/v3.5.14/etcd-v3.5.14-darwin-amd64.zip`, and unpack it, the `go version` result corroborates that it finds 0.17.0.

![image](https://github.com/etcd-io/etcd/assets/96485235/60e4561b-0f5f-46de-933b-33fd5a9f95ab)

Am I missing anything here?   Please help.   Thanks.

**Comment by charlie-ang-collibra** [comment]
Hi @henrybear327, I'm hoping you might be able to help me confirm if this is just an error on my side, or if the CVE is still open?

It looks like the patch was only made to the main `go.mod`, while the corresponding `go.mod` files under `client`, `etcdutl` and `etcdctl` remained unpatched?

Thank you so much in advance.

**Comment by henrybear327** [comment]
@ahrtr I think I need to address 2 issues (and I am on it now)
- patch the missed updates
- patch the CI regarding the vulnerability checker

Based on the grep log I indeed missed other `go.mod` files, and the CI also didn't catch this...
```
➜  etcd git:(release-3.5) ✗ grep -Ri "golang.org/x/net v" | grep -v sum
./etcdutl/go.mod:       golang.org/x/net v0.17.0 // indirect
./go.mod:       golang.org/x/net v0.23.0 // indirect
./tools/mod/go.mod:     golang.org/x/net v0.0.0-20201021035429-f5854403a974 // indirect
./etcdctl/go.mod:       golang.org/x/net v0.17.0 // indirect
./tests/go.mod: golang.org/x/net v0.17.0 // indirect
./server/go.mod:        golang.org/x/net v0.17.0
./api/go.mod:   golang.org/x/net v0.17.0 // indirect
./client/v3/go.mod:     golang.org/x/net v0.17.0 // indirect
./pkg/go.mod:   golang.org/x/net v0.17.0 // indirect
```

**Reply by charlie-ang-collibra** [reply]
Thank you, @henrybear327.   I wasn't sure if this was the right way to ask the question.  🙏

**Reply by henrybear327** [reply]
Thanks for flagging this @charlie-ang-collibra :)  
I wasn't aware that I missed those and the CI also missed these.

**Reply by charlie-ang-collibra** [reply]
No worries.  We're always happy for your support on etcd!

**Reply by henrybear327** [reply, marked as answer]
Initial PR for release branch 3.5 regarding CI and dependency updates is up for review!

**Reply by ahrtr** [reply]
> @ahrtr I think I need to address 2 issues (and I am on it now)
> 
> * patch the missed updates
> * patch the CI regarding the vulnerability checker
> 
> Based on the grep log I indeed missed other `go.mod` files, and the CI also didn't catch this...
> 
> ```
> ➜  etcd git:(release-3.5) ✗ grep -Ri "golang.org/x/net v" | grep -v sum
> ./etcdutl/go.mod:       golang.org/x/net v0.17.0 // indirect
> ./go.mod:       golang.org/x/net v0.23.0 // indirect
> ./tools/mod/go.mod:     golang.org/x/net v0.0.0-20201021035429-f5854403a974 // indirect
> ./etcdctl/go.mod:       golang.org/x/net v0.17.0 // indirect
> ./tests/go.mod: golang.org/x/net v0.17.0 // indirect
> ./server/go.mod:        golang.org/x/net v0.17.0
> ./api/go.mod:   golang.org/x/net v0.17.0 // indirect
> ./client/v3/go.mod:     golang.org/x/net v0.17.0 // indirect
> ./pkg/go.mod:   golang.org/x/net v0.17.0 // indirect
> ```

Thanks @henrybear327 for the investigation. 

Human make mistakes. I think we should also consider enhancing our workflow to automatically identify such inconsistencies.

**Reply by henrybear327** [reply]
> > @ahrtr I think I need to address 2 issues (and I am on it now)
> > 
> > * patch the missed updates
> > * patch the CI regarding the vulnerability checker
> > 
> > Based on the grep log I indeed missed other `go.mod` files, and the CI also didn't catch this...
> > ```
> > ➜  etcd git:(release-3.5) ✗ grep -Ri "golang.org/x/net v" | grep -v sum
> > ./etcdutl/go.mod:       golang.org/x/net v0.17.0 // indirect
> > ./go.mod:       golang.org/x/net v0.23.0 // indirect
> > ./tools/mod/go.mod:     golang.org/x/net v0.0.0-20201021035429-f5854403a974 // indirect
> > ./etcdctl/go.mod:       golang.org/x/net v0.17.0 // indirect
> > ./tests/go.mod: golang.org/x/net v0.17.0 // indirect
> > ./server/go.mod:        golang.org/x/net v0.17.0
> > ./api/go.mod:   golang.org/x/net v0.17.0 // indirect
> > ./client/v3/go.mod:     golang.org/x/net v0.17.0 // indirect
> > ./pkg/go.mod:   golang.org/x/net v0.17.0 // indirect
> > ```
> 
> Thanks @henrybear327 for the investigation.
> 
> Human make mistakes. I think we should also consider enhancing our workflow to automatically identify such inconsistencies.

Thanks @ahrtr for the kind words!

I will take care of the enhancement with @ivanvc

**Reply by charlie-ang-collibra** [reply]
Thank you for being super-responsive!

**Reply by ivanvc** [reply]
I created the issues #18180 and #18173 to track the enhancements. Henry solved the dependency issue with pull request #18170.

**Reply by charlie-ang-collibra** [reply]
Hi!  How do I watch for news about the next release window for 3.5?

**Reply by ivanvc** [reply]
You can sign up to the etcd.io dev mailing list. Refer to https://etcd.io/community/, https://groups.google.com/g/etcd-dev.

**Reply by charlie-ang-collibra** [reply]
Hi @ivanvc .  Nice to see you virtually yesterday!    

I just wanted to pass along that even though CVE-2023-45288 [has no official NVD CVSS](https://nvd.nist.gov/vuln/detail/CVE-2023-45288) score, Snyk classifies it as a [7.5](https://security.snyk.io/vuln/SNYK-GOLANG-GOLANGORGXNETHTTP2-6531285).   Does the maintainer community rely exclusively on NVD scoring?

**Reply by ahrtr** [reply]
I think we should release 3.5.15, since go 1.21.11 also includes CVE fixes. cc @jmhbnz @ivanvc

**Reply by serathius** [reply]
Same applies to v3.4.33?

**Reply by ahrtr** [reply]
yes, both 3.5.15 and 3.4.33.

**Reply by jmhbnz** [reply]
Agreed our patch release criteria has been met so we should release `3.5.15` and `3.4.34` - I'll create release planning issues shortly.

**Reply by ivanvc** [reply]
I created a planning issue for 3.5.15: #18247.

However, we don't have any outstanding work for the 3.4 release. The 3.5 release criteria is met because of https://github.com/etcd-io/etcd/pull/18170, but 3.4 (AFAIK) doesn't present that vulnerability (correct me if I'm wrong).

**Reply by ivanvc** [reply]
Also, 3.4.33 was released with Go v1.21.11 (#18130) addressing CVE 2024-24790. But 3.5.14 didn't, as it was released weeks before.

**Reply by jmhbnz** [reply]
Good catch @ivanvc - `release-3.4` should be ok for now then. Let's push ahead with `3.5.15` only.

**Reply by charlie-ang-collibra** [reply]
The progress towards release is very much appreciated.  Thank you.  🙏

**Reply by ivanvc** [reply]
Hi team, just updating as we released [3.5.15](https://github.com/etcd-io/etcd/releases/tag/v3.5.15) earlier today, which contains these fixes.

**Reply by charlie-ang-collibra** [reply]
Thank you so much for the update, @ivanvc !  Have a great weekend!
